### ISSUE TITLE
[Component] [sub-menu] ElSubMenu在nuxt@3.0.0下不可用

### ISSUE THREAD

**Issue by MZ-Dlovely** [element-plus/element-plus#11476]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.45`
- Element Plus Version: `2.2.17`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36 Edg/109.0.1518.78`
- Build Tool: `Nuxt`

### Related Component

- `el-sub-menu`

### Reproduction Link

[Github Repo](https://github.com/miaomiaoTeam/miaoNuggetsWebsite/tree/feat-account_auth-miao-dev)

### Steps to reproduce

1. 启动服务
  ```shell
  $ pnpm i
  $ pnpm dev
  ```

2. 打开网页
  http://localhost:3000/_admin

### What is Expected?

正常显示可展开菜单

### What is actually happening?

[Vue warn]: Failed to resolve component: ElSubMenu
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.

### Additional comments

`node_modules/element-plus/es/component.mjs`里未导入`ElSubMenu`并集合导出.(line 36)
`node_modules/@element-plus/nuxt/dist/module.mjs`里引用的是`node_modules/element-plus/es/component.mjs`默认导出的`Components`集合.(line 2)

另外，`node_modules/element-plus/global.d.ts`里有导出`ElSubMenu`类型所以编辑器未报错

1. 报错信息

<img width="351" alt="image" src="https://user-images.githubusercontent.com/93523064/217605875-53182986-677a-4e51-a3d9-2b36163101d5.png">

2. 相关导入

- `node_modules/@element-plus/nuxt/dist/module.mjs`(line 2)
  <img width="213" alt="image" src="https://user-images.githubusercontent.com/93523064/217606078-425bd046-6dfc-455f-8986-453892482e40.png">
- `node_modules/element-plus/es/component.mjs`(line 36)
  <img width="319" alt="image" src="https://user-images.githubusercontent.com/93523064/217606130-8208d103-bc06-4d06-96ea-cbff9fd6f589.png">

**Comment by MZ-Dlovely** [comment]
# 目前的缓解方案是这样

```ts
// modules/element-plus-component.ts
import { addComponent, defineNuxtModule } from '@nuxt/kit'
import { ElSubMenu } from 'element-plus'

export default defineNuxtModule({
  meta: {
    name: 'element-plus-components',
    configKey: 'elementPlusComponents',
  },
  defaults: {},
  setup() {
    const components = new Set([ElSubMenu])
    for (const { name } of components) {
      console.log(name)
      addComponent({
        export: name,
        name,
        filePath: 'element-plus',
      })
    }
  },
})
```

**Comment by tolking** [comment]
Fixed in #11100, you can update Element
 Plus to version 2.2.28 and above.

Or refer to the [doc](https://github.com/element-plus/element-plus-nuxt#components) to modify the config of components in nuxt.config.ts.
